### ISSUE TITLE
Figures are not currently being rendered in the docs

### ISSUE THREAD

**Issue by jni** [jni/skan#146]
See e.g.:

https://jni.github.io/skan/complete_analysis.html

**Comment by lucyleeow** [comment]
Sorry, silly question, which part of `complete_analysis.rst` should be figures...?

**Comment by jni** [comment]
The very last cell.

The FAQ also is supposed to show a figure. (Less code to trawl through!)

**Comment by lucyleeow** [comment]
Ah I didn't know about the `nbplot` directive. Will look into it!

**Comment by jni** [comment]
It's provided by the nb2plots sphinx extension. https://github.com/matthew-brett/nb2plots

I wonder if it broke with a recent sphinx release?

Ultimately I was considering moving to MyST for those pages, so maybe now is the right time to do that...

**Comment by lucyleeow** [comment]
Trying to make the docs and looking at `complete_analysis` - where is the `schizonts` directory..?

```Python
schizont_files = glob('schizonts/*.tif')
```

Another silly question, I'm guessing readthedocs uses `test-environment.yml` env to create the docs. Is there a `test-environment.yml` file?

**Comment by jni** [comment]
> Trying to make the docs and looking at `complete_analysis` - where is the `schizonts` directory..?

I guess you missed the [top of the notebook](https://jni.github.io/skan/complete_analysis.html#getting-the-data):

> First, we need to download the input data, which is posted at the Open Science Framework (OSF) as a zip file. Many thanks to the OSF for providing hosting for scientific data. The data be downloaded at this URL:
> `https://osf.io/7vyx3`
> Download and extract the zipfile, and make sure the `schizonts` folder contained within it is placed in this notebook’s working directory.

Yes I am aware that it would be best if this process was automated. 😜 

> I'm guessing readthedocs uses test-environment.yml env to create the docs. Is there a test-environment.yml file?

Oops! So, we are not hosted by RTD anymore, that file is a holdover from the earliest days of skan. It can be deleted, just like the test-environment.yml file was deleted.

You should be looking at the [build-docs](https://github.com/jni/skan/blob/c9b712382968fadaa7c1eeaa3a0bed5ab4417cda/.github/workflows/build-docs.yml) and [deploy-docs](https://github.com/jni/skan/blob/main/.github/workflows/deploy-docs.yml) github actions.

**Comment by lucyleeow** [comment]
I had missed a step here. I have found `build-docs.yml` and see how it works now! 😬

Running it locally, `getting_started` seems to work and generates images in `doc/_build/nbplot_directive`. There was an exception with running `complete_analysis`:
```Python
>>> ridges = ((data['mean shape index'] < 0.625) &
...           (data['mean shape index'] > 0.125))

Exception:
Traceback (most recent call last):
  File "/home/<user>/miniconda3/envs/skan/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 3361, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 76, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 108, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 5198, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 5206, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'mean shape index'
```
Checking, `data` does not seem to have that column? Could the data have changed?
```
In [6]: data.columns
Out[6]: 
Index(['skeleton-id', 'node-id-src', 'node-id-dst', 'branch-distance',
       'branch-type', 'mean-pixel-value', 'stdev-pixel-value',
       'image-coord-src-0', 'image-coord-src-1', 'image-coord-dst-0',
       'image-coord-dst-1', 'coord-src-0', 'coord-src-1', 'coord-dst-0',
       'coord-dst-1', 'euclidean-distance', 'squiggle', 'scale', 'filename'],
      dtype='object')
```

This was the error in `faq.rst`:
```Python
>>> for ax, mode in zip(axes.ravel(), ('none', 'centroid', 'mst')):
...     graph, coordinates = skeleton_to_csgraph(skel, junction_mode=mode)
...     draw.overlay_skeleton_networkx(
...             graph, coordinates, image=skel, axis=ax
...             )
...     ax.set_title(f'junctions: {mode}')

Exception:
Traceback (most recent call last):
  File "/home/<user>/miniconda3/envs/skan/lib/python3.9/site-packages/nb2plots/nbplots.py", line 901, in run_code
    six.exec_(code, ns)
  File "<string>", line 12, in <module>
TypeError: skeleton_to_csgraph() got an unexpected keyword argument 'junction_mode'
```

Maybe the example was from an older version of skan?

If it is of interest, you could test doc building in CI? Note would have to set warning to be errors as exceptions raised during execution are warnings, e.g.:
```
/home/<user>/skan/doc/faq.rst:40: WARNING: Exception plotting faq-1
```

**Comment by jni** [comment]
> If it is of interest, you could test doc building in CI? Note would have to set warning to be errors as exceptions raised during execution are warnings, e.g.:

Ha! I *thought* I was testing these things! This probably explains everything: at some point, all of these things broke, but we didn't catch them because I wasn't actually testing the running of the stuff.

The FAQ case is interesting and terrifying. :stuck_out_tongue_winking_eye: It actually requires an old version of skan to work (0.10) precisely because in 0.11+ we will remove thta functionality. Perhaps we prebuild the figure using 0.10 and don't run that file? Any thoughts on the best way to do this? Happy to hop on a call to discuss.

I think I'd like to stop depending on nbplots and instead port them all to use jupytext, jupyterbook, and nb-myst. Do you think you could handle that transition?

**Comment by smith6jt-cop** [comment]
Just so you don't have the big ugly error on your website, you could change

`cellmeans = (datar.groupby(['infection', 'cell number'])
                  .mean().reset_index())`

to

`cellmeans = datar.groupby(['infection', 'cell number'])
cellmeans = cellmeans['branch distance (nm)'].mean().reset_index()`

**Comment by smith6jt-cop** [comment]
![image](https://github.com/jni/skan/assets/73438963/98f40126-c39e-4385-84b1-cb73561c7522)

**Comment by jni** [comment]
Thanks for the ping @smith6jt-cop! Actually this is fixed in main, we just haven't released since we fixed it:

https://skeleton-analysis.org/dev/examples/complete_analysis.html#making-the-figure

See source at:

https://github.com/jni/skan/blob/b703cf603f34bff03971cfce1d156f0160206a23/doc/examples/complete_analysis.md#3-making-the-figure

In this case we fixed by using `mean(numeric_only=True)`.

Need to make a release but just a bit swamped. 🙏

**Comment by jni** [comment]
Fixed by #221
